cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/update/StateVariablesExport.java

```diff
@@ -0,0 +1,408 @@
+            Bus b = dl.getTerminal().getBusBreakerView().getBus();
+            Optional<String> topologicalNode = dl.getAliasFromType(CGMES_PREFIX_ALIAS + CgmesNames.TOPOLOGICAL_NODE);
+            if (topologicalNode.isPresent()) {
+                if (dl.hasProperty("v") && dl.hasProperty("angle")) {
+                    writeVoltage(topologicalNode.get(), Double.valueOf(dl.getProperty("v", "NaN")), Double.valueOf(dl.getProperty("angle", "NaN")), writer);
+                } else if (b != null) {
+                    // calculate complex voltage value: abs for VOLTAGE, degrees for ANGLE
+                    Complex v2 = complexVoltage(dl.getR(), dl.getX(), dl.getG(), dl.getB(), b.getV(), b.getAngle(), dl.getP0(), dl.getQ0());
+                    writeVoltage(topologicalNode.get(), v2.abs(), Math.toDegrees(v2.getArgument()), writer);
+                } else {
+                    writeVoltage(topologicalNode.get(), 0.0, 0.0, writer);
+                }
+            }
+        }
+        // Voltages at inner nodes of Tie Lines
+        // (boundary nodes that have been left inside CGM)
+        for (Line l : network.getLines()) {
+            if (!l.isTieLine()) {
+                continue;
+            }
+            TieLine tieLine = (TieLine) l;
+            // FIXME(Luma) Obtain voltage at inner node
+            LOG.error("Must export topologicalNode voltage for boundary Tie Line {}", tieLine);
+        }
+    }
+
+    private static void writeVoltage(String topologicalNode, double v, double angle, XMLStreamWriter writer) throws XMLStreamException {
+        writer.writeStartElement(CIM_NAMESPACE, "SvVoltage");
+        writer.writeAttribute(RDF_NAMESPACE, ID, getUniqueId());
+        writer.writeStartElement(CIM_NAMESPACE, SV_VOLTAGE_ANGLE);
+        writer.writeCharacters(fs(angle));
+        writer.writeEndElement();
+        writer.writeStartElement(CIM_NAMESPACE, SV_VOLTAGE_V);
+        writer.writeCharacters(fs(v));
+        writer.writeEndElement();
+        writer.writeEmptyElement(CIM_NAMESPACE, SV_VOLTAGE_TOPOLOGICAL_NODE);
+        writer.writeAttribute(RDF_NAMESPACE, CgmesNames.RESOURCE, "#" + topologicalNode);
+        writer.writeEndElement();
+    }
+
+    private static void writePowerFlows(Network network, XMLStreamWriter writer) {
+        writeInjectionPowerFlowToCgmes(network, writer, Network::getLoadStream);
+        writeInjectionPowerFlowToCgmes(network, writer, Network::getGeneratorStream);
+        writeInjectionPowerFlowToCgmes(network, writer, Network::getShuntCompensatorStream);
+        writeInjectionPowerFlowToCgmes(network, writer, Network::getStaticVarCompensatorStream);
+        writeInjectionPowerFlowToCgmes(network, writer, Network::getBatteryStream);
+
+        network.getDanglingLineStream().forEach(dl -> dl.getAliasFromType(CGMES_PREFIX_ALIAS + "Terminal_Boundary")
+                .ifPresent(terminal -> writePowerFlow(terminal, dl.getP0(), dl.getQ0(), writer)));
```

Comment:
If you want to write here the updated values of P and Q at boundary point, I think that you cannot make it like that. In IIDM, after a power flow calculation, the only think that changes is P and Q of the terminal connected to the network. It means that we need to introduce new properties to be set by the power flow engine or re-computed them. This value will be use to update EquivalentInjection too. @zamarrenolm do you confirm?

Reply:
And it will solve the question below: and what about danglignq line with generation part?

Reply:
You're right, I think this is only valid if generation is zero... @zamarrenolm Can you confirm?

Reply:
The flow at the boundary terminal is equal to the generation of the dangling line, but with opposite sign (`p0`, `q0` are the values of the injection, with load sign convention, and the flow in the dangling line from the boundary to the network side is the same quantity, also with load sign convention)

Reply:
Update: after a powerflow calculation, P0 and Q0 are considered as targets and are not updated in IIDM. But, after a balance ajustement, it should be updated and so, after topological merge + balance ajustement, these lines of codes will provide updated values.

Reply:
After discussing it,  it seems a mistake to export `p0` and `q0` as calculated flows. We should introduce an object called `middleBus` or something which will contain the calculated flows, voltage and angle of the boundary fictitious terminal of the dangling line. For now, I think we should delete this line and put a `TODO`

Reply:
(I did not delete the line but put a `FIXME`)